tests/testsuite/test.rs

```diff
@@ -3658,3 +3658,160 @@ fn test_dep_with_dev() {
         )
         .run();
 }
+
+#[cfg(target_env = "gnu")]
+#[cargo_test]
+fn cargo_test_doctest_xcompile_ignores() {
+    use crate::support::is_nightly;
+    if !is_nightly() {
+        return;
+    }
+    let p = project()
+        .file("Cargo.toml", &basic_lib_manifest("foo"))
+        .file(
+            "src/lib.rs",
+            r#"
+
+            ///```ignore-gnu
+            ///assert!(1 == 1);
+            ///```
+            pub fn foo() -> u8 {
+                4
+            }
+            "#,
+        )
+        .build();
+
+    p.cargo("build").run();
+    p.cargo("test")
+        .with_stdout_contains(
+            "\
+             test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out\
+             ",
+        )
+        .run();
+    p.cargo("test -Zdoctest-xcompile")
+        .masquerade_as_nightly_cargo()
+        .with_stdout_contains(
+            "\
+             test result: ok. 0 passed; 0 failed; 1 ignored; 0 measured; 0 filtered out\
+             ",
+        )
+        .run();
+}
+
+#[cfg(target_env = "gnu")]
```

Comment:
Like below I don't think this is gnu-specific, so this would be good to generalize to run on all platforms.

Reply:
By that do you mean all platform used in cargo ci? Since I need to specify a target that is both installed and that can run on the current machine, it would be straightforward to extend the test to include both apple-darwin and pc-windows-msvc targeting their i686 variants, but I am not sure what I would do in the general case, short of just using the same target as host

Reply:
Yes there's already a number of tests that do cross compilation, so you can mirror those to find the "other target", and that's all configured on CI as well.